cli/cmd/profile.go

```diff
@@ -118,11 +147,121 @@ Example:
 
 	cmd.PersistentFlags().BoolVar(&options.template, "template", options.template, "Output a service profile template")
 	cmd.PersistentFlags().StringVar(&options.openAPI, "open-api", options.openAPI, "Output a service profile based on the given OpenAPI spec file")
+	cmd.PersistentFlags().StringVar(&options.tap, "tap", options.tap, "Output a service profile based on tap data for the given target resource")
+	cmd.PersistentFlags().StringVarP(&options.tapDuration, "tap-duration", "t", options.tapDuration, "Duration over which tap data is collected (for example: \"10s\", \"1m\", \"10m\")")
 	cmd.PersistentFlags().StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace of the service")
 
 	return cmd
 }
 
+func renderTapOutputProfile(options *profileOptions, controlPlaneNamespace string, w io.Writer) error {
+	profile := sp.ServiceProfile{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s.%s.svc.cluster.local", options.name, options.namespace),
+			Namespace: controlPlaneNamespace,
+		},
+		TypeMeta: meta_v1.TypeMeta{
+			APIVersion: "linkerd.io/v1alpha1",
+			Kind:       "ServiceProfile",
+		},
+	}
+
+	log.Debug("Running `linkerd tap %s --namespace %s`", options.tap, options.namespace)
+	client := cliPublicAPIClient()
+	requestParams := util.TapRequestParams{
+		Resource:  options.tap,
+		Namespace: options.namespace,
+	}
+	req, err := util.BuildTapByResourceRequest(requestParams)
```

Comment:
should this filter on `proxy=in`? another way to say it, should it do this:
```bash
bin/linkerd tap deploy --to deploy/web -n emojivoto
```

when i generate a profile from `deploy/web`, i'm getting expected routes like `/api/list`, but i'm also getting `/emojivoto.v1.EmojiService/ListAll`, which i wouldn't expect on `web.emojivoto.svc.cluster.local`:
```bash
$ bin/go-run cli profile web --tap deploy/web -n emojivoto --tap-duration 2s
apiVersion: linkerd.io/v1alpha1
kind: ServiceProfile
metadata:
  creationTimestamp: null
  name: web.emojivoto.svc.cluster.local
  namespace: linkerd
spec:
  retryBudget: null
  routes:
  - condition:
      method: GET
      pathRegex: /api/list
    name: GET /api/list
  - condition:
      method: GET
      pathRegex: /api/vote
    name: GET /api/vote
  - condition:
      method: POST
      pathRegex: /emojivoto\.v1\.EmojiService/FindByShortcode
    name: POST /emojivoto.v1.EmojiService/FindByShortcode
  - condition:
      method: POST
      pathRegex: /emojivoto\.v1\.EmojiService/ListAll
    name: POST /emojivoto.v1.EmojiService/ListAll
  - condition:
      method: POST
      pathRegex: /emojivoto\.v1\.VotingService/VoteGirl
    name: POST /emojivoto.v1.VotingService/VoteGirl
  - condition:
      method: POST
      pathRegex: /emojivoto\.v1\.VotingService/VotePoop
    name: POST /emojivoto.v1.VotingService/VotePoop
```

Reply:
ahhhh. I think that makes sense.

Reply:
we'd run into the problem of missing traffic between different resource types though.
what if we did `linkerd tap namespace --to deploy/web -n emojivoto` instead?

thoughts? cc @adleong

Reply:
I think we want to look at inbound traffic for this, not outbound.  Is it possible to still use `linkerd tap deploy/web` but post-filter the results to only consider proxy=in?